### PR TITLE
feat(pennylane): Make penny lane syncs and actions into public templates

### PR DESCRIPTION
**Pull request by dannylwe**
## Describe your changes
- add actions for pennylane (customer, supplier, invoice, supplier)
- add syncs for pennylane (customer, supplier, invoice, supplier)

~~Note: syncs tests are currently failing.~~
## Checklist before requesting a review (skip if just adding/editing APIs & templates)

-   [x] I added tests, otherwise the reason is:
-   [x] External API requests have `retries`
-   [x] Pagination is used where appropriate
-   [x] The built in `nango.paginate` call is used instead of a `while (true)` loop
-   [x] Third party requests are NOT parallelized (this can cause issues with rate limits)
-   [ ] If a sync requires metadata the `nango.yaml` has `auto_start: false`
-   [ ] If the sync is a `full` sync then `track_deletes: true` is set
-   [x] I followed the best practices and guidelines from the [Writing Integration Scripts](/NangoHQ/integration-templates/blob/main/WRITING_INTEGRATION_SCRIPTS.md) doc

### CODE REVIEW

File: integrations/pennylane/syncs/suppliers.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import type { CreateSupplier, NangoSync, PennylaneSupplier, ProxyConfiguration } from '../../models.js';
+import { toSupplier } from '../helpers.js';
+
+export default async function fetchData(nango: NangoSync) {
+    const config: ProxyConfiguration = {
+        // https://pennylane.readme.io/reference/suppliers-post
```
Comment (khaliqgant) [inline integrations/pennylane/syncs/suppliers.ts:6]: Documentation should be to https://pennylane.readme.io/reference/suppliers-get

File: integrations/pennylane/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    CreateInvoice,
+    CreateSupplier,
+    NangoAction,
+    PennylaneCustomer,
+    PennylaneIndividualCustomer,
+    PennylaneInvoice,
+    PennylaneSupplier
+} from '../models.js';
+
+export function toInvoice(invoice: PennylaneInvoice): PennylaneInvoice {
```
Comment (khaliqgant) [inline integrations/pennylane/helpers.ts:11]: for consistency purposes these mapping functions should go in specific files

`mappers/to-invoice.ts`
`mappers/to-customer.ts`
`mappers/to-supplier.ts`

File: integrations/pennylane/syncs/suppliers.ts (hunk 1 above)
Comment (dannylwe) [reply integrations/pennylane/syncs/suppliers.ts]: this has been updated

**Review by khaliqgant — CHANGES_REQUESTED** [review]
Can you add the products sync and action as well to this?

File: integrations/pennylane/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import type { CreateInvoice, NangoAction } from '../models.js';
+
+export function createInvoice(input: CreateInvoice) {
```
Comment (khaliqgant) [inline integrations/pennylane/helpers.ts:3]: move to a mappers file and type the return output

File: integrations/pennylane/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                    imputation_dates: {
+                        start_date: new Date(input.imputation_dates.start_date).toISOString(),
+                        end_date: new Date(input.imputation_dates.end_date).toISOString()
+                    }
+                })
+        }
+    };
+}
+
+export function validateInvoiceInput(input: CreateInvoice, nango: NangoAction) {
```
Comment (khaliqgant) [inline integrations/pennylane/helpers.ts:40]: Couldn't instead use zod for this?

**Review by khaliqgant — APPROVED** [review]

**Pull request merged**